google-cloud-spanner/src/main/java/com/google/cloud/spanner/BatchClientImpl.java

```diff
@@ -59,6 +61,12 @@ public class BatchClientImpl implements BatchClient {
   @GuardedBy("multiplexedSessionLock")
   private final AtomicReference<SessionImpl> multiplexedSessionReference;
 
+  /**
+   * This flag is set to true if the server return UNIMPLEMENTED when partitioned transaction is
+   * executed on a multiplexed session. TODO: Remove once this is guaranteed to be available.
+   */
+  @VisibleForTesting final AtomicBoolean unimplementedForPartitionedOps = new AtomicBoolean(false);
```

Comment:
Can/should we make this a static field, so that it is shared across all batch clients? I assume that once this flag is set, then we should just fall back to regular sessions for all clients, and not first try multiplexed sessions for each new batch client.